test/unit/math/binaryOps/src/binaryOps.cpp

```diff
@@ -0,0 +1,432 @@
+//
+// Created by jakob on 19.08.19.
+//
+
+#include <catch2/catch.hpp>
+
+#include <alpaka/alpaka.hpp>
+#include <alpaka/test/MeasureKernelRunTime.hpp>
+#include <alpaka/test/acc/Acc.hpp>
+#include <alpaka/test/queue/Queue.hpp>
+
+#include <iostream>
+#include <typeinfo>
+#include <cmath>
+
+/*
+ * atan2     | Y | R^2\{(0,0)}
+ * fmod      | Y | R^2\{(x,0)|x in R}
+ * max       | Y | R^2
+ * min       | Y | R^2
+ * remainder | Y | R^2\{(x,0)|x in R}
+ * sincos    | Y | R^2 -> return 2 R
+*/
+
+/*
+ * if you want to add a new operation simply add it to the array.
+ * 1. Specify the std::  implementation.
+ * 2. If the std function has a const reference signature use this
+ * 3. Specify the alpaka implementation.
+ * 4. Define the range in which the operator should be testes against.
+ */
+
+/*
+ * if you need to add a new range you have to add it to the switch case
+ *  - in the kernel class
+ *  - in the TestTemplate
+ */
+
+
+
+
+// possible definition ranges
+enum class Range
+{
+    POSITIVE_ONLY,
+    NOT_ZER0,
+    X_NOT_ZERO,
+    Y_NOT_ZERO,
+    UNRESTRICTED
+};
+
+// function pointer templates for std::math and alpaka::math
+template<
+        typename TAcc,
+        typename T>
+using alpaka_func_ptr = T (*) (TAcc const & , T const &, T const &);
+
+template<
+        typename T>
+using std_func_ptr = T (*) (T, T);
+
+// used for all operators that need const references (like min())
+template<
+        typename T>
+using std_func_ptr_const = T const & (*) (T const & , T const &);
+
+
+template<
+        typename TAcc,
+        typename T>
+struct TestStruct
+{
+    std_func_ptr<T> stdOp ;
```

Comment:
can you add some documentation e.g. that `stdOp` can be a `nullptr` and the behavior in that case?